### PR TITLE
Update reporttheme output to remove hard coded colour names

### PR DESCRIPTION
**Pull request by asears**
## PR Summary

Remove hard-coded color names from Output.ps1. Introduce PR for suggestions of pluggable theme & splitting up failure color with failure message color. #1735 relates.

Previously https://github.com/pester/Pester/pull/1752

## PR Checklist

- [x] PR has meaningful title
- [x] Summary describes changes
- [x] PR is ready to be merged
  - If not, use the arrow next to `Create Pull Request` to mark it as a draft. PR can be marked `Ready for review` when it's ready.
- [] Tests are added/update *(if required)*
- [ ] Documentation is updated/added *(if required)*

### CODE REVIEW

File: src/functions/Output.ps1 [hunk 1]
```diff
@@ -57,6 +58,10 @@ $script:ReportTheme = DATA {
         Foreground       = 'White'
         Information      = 'DarkGray'
         Coverage         = 'White'
+        CoverageWarn     = 'DarkRed'
```
Comment (nohwnd) [inline src/functions/Output.ps1:61]: This does not seem to be used. If it is used and I just don't see it, warning should be yellow probably.

Comment (nohwnd) [reply src/functions/Output.ps1:61]: @asears ping on this.

Comment (asears) [reply src/functions/Output.ps1]: Makes sense, removed as not currently in use.

File: src/functions/Output.ps1 [hunk 2]
```diff
@@ -424,15 +428,13 @@ function ConvertTo-FailureLines {
                 [String]$isShould = '^at (Should<End>|Invoke-Assertion), .*\\Pester.psm1: line [0-9]*$'
             }
 
-            # PESTER_BUILD
-            if ($true) {
+            if ($true) { # PESTER_BUILD
```
Comment (nohwnd) [inline src/functions/Output.ps1:431]: Please revert this change. This would leave `if ($true) {` in place.

File: src/functions/Output.ps1 [hunk 3]
```diff
@@ -492,7 +494,7 @@ function Get-WriteScreenPlugin ($Verbosity) {
 
         if ($PesterPreference.Output.Verbosity.Value -in 'Detailed', 'Diagnostic') {
             $activeFilters = $Context.Filter.psobject.Properties | & $SafeCommands['Where-Object'] { $_.Value }
-            if ($null -ne $activeFilters) {
+            if($null -ne $activeFilters) {
```
Comment (nohwnd) [inline src/functions/Output.ps1:497]: nit: space

File: src/functions/Output.ps1 [hunk 4]
```diff
@@ -648,15 +650,15 @@ function Get-WriteScreenPlugin ($Verbosity) {
 
             Failed {
                 # If VSCode and not Integrated Terminal (usually a test-task), output Pester 4-format to match 'pester'-problemMatcher in VSCode.
-                if ($env:TERM_PROGRAM -eq 'vscode' -and -not $psEditor) {
+                if($env:TERM_PROGRAM -eq 'vscode' -and -not $psEditor) {
```
Comment (nohwnd) [inline src/functions/Output.ps1:653]: nit: space

File: src/functions/Output.ps1 [hunk 5]
```diff
@@ -648,15 +650,15 @@ function Get-WriteScreenPlugin ($Verbosity) {
...
 
                     # Loop to generate problem for every failed assertion per test (when $PesterPreference.Should.ErrorAction.Value = "Continue")
-                    foreach ($e in $_test.ErrorRecord) {
+                    foreach($e in $_test.ErrorRecord) {
```
Comment (nohwnd) [inline src/functions/Output.ps1:656]: nit: space

File: src/functions/Output.ps1 (hunk 3 above)
Comment (nohwnd) [reply src/functions/Output.ps1:497]:
```suggestion
            if ($null -ne $activeFilters) {
```

File: src/functions/Output.ps1 (hunk 4 above)
Comment (nohwnd) [reply src/functions/Output.ps1:653]:
```suggestion
                if ($env:TERM_PROGRAM -eq 'vscode' -and -not $psEditor) {
```

File: src/functions/Output.ps1 (hunk 5 above)
Comment (nohwnd) [reply src/functions/Output.ps1:656]:
```suggestion
                    foreach ($e in $_test.ErrorRecord) {
```

**Comment by nohwnd** [comment]
@asears looks done, are you done? :)

**Comment by nohwnd** [comment]
@fflaten maybe you'd be interesting in fixing the merge conflict and merging this? I think you should have enough permissions for this. :)

**Comment by fflaten** [comment]
Done. Took the liberty of switching Container vs Discovery some places and fixing some hard-coded colors that remaining, ex. the new pester version message.

**Comment by fflaten** [comment]
Merging PR is still up to you @nohwnd since main is protected.
